### PR TITLE
Product Gallery block: Fix incorrect icon color when block is selected

### PR DESCRIPTION
**Pull request by thealexandrelara**
Currently, when selecting the Large Image block, the icon color remains the same as when the block is not selected. The expected behavior is that when the block is selected, the icon color should reflect the selected state (shown in white color).

# Walkthrough

### <samp>🤖 Generated by Copilot at b9eb7b0</samp>

*  Update `Icon` component to use `currentColor` and improve design ([link](https://github.com/woocommerce/woocommerce-blocks/pull/10426/files?diff=unified&w=0#diff-fa3290413ea65e343292780651de103e95d8d529c85c19c452df77b669482cecL2-R32))

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR has either a `[type]` label or a `[skip-changelog]` label.
- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Screenshots

| Before | After |
| ------ | ----- |
| image  | image  |

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Log in to your WordPress dashboard.
2. From your WordPress dashboard, go to Appearance > Themes. Make sure you have a block-based theme installed and activated. If not, you can install one from the Add New option. Block-based themes include "Twenty-twenty Two," "Twenty-twenty Three," etc.
3. On the left-hand side menu, click on Appearance > Editor > Templates
5. Find and select the 'Single Product' template from the list.
6. When the Classic Product Template renders, click on Transform into Blocks. This will transform the Classic template in a block template if you haven't done it before.
7. On the top-right side, click on the Save button.
8. Inside the Site editor, click on the '+' button to add a new block. In the block library that pops up, search for the 'Product Gallery' block.
9. Click on the 'Product Gallery' block to add it to the template. 
10. On the left side, check the List View and select the Product Gallery block. Make sure the icon changes it's color from a dark color to a white color when the block is selected on the list

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

### CODE REVIEW

**Review by danieldudzic — CHANGES_REQUESTED** [review]
This does fix the color changing problem, but there seems to be a slight alignment issue:

| Before | After |
| ------ | ----- |
|   Single_Product_‹_Template_‹_Editor_‹_productgallery_—_WordPress-3 | Single_Product_‹_Template_‹_Editor_‹_productgallery_—_WordPress-2 |

**Comment by thealexandrelara** [comment]
> This does fix the color changing problem, but there seems to be a slight alignment issue:

Nice catch! I just fixed it, so I think the alignment is now correct

**Review by danieldudzic — CHANGES_REQUESTED** [review]
Almost! 😅 

Single_Product_‹_Template_‹_Editor_‹_productgallery_—_WordPress

File: assets/js/blocks/product-gallery/icon.tsx [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
-		</mask>
 		<rect
-			width="6"
-			height="5.5"
-			x="4"
-			y="14.5"
-			stroke="#000"
-			strokeWidth="3"
-			mask="url(#a)"
+			x="5"
```
Comment (danieldudzic) [inline assets/js/blocks/product-gallery/icon.tsx]:
```suggestion
			x="4.75"
```

File: assets/js/blocks/product-gallery/icon.tsx [hunk 2]
```diff
@@ -1,35 +1,34 @@
...
-		</mask>
 		<rect
-			width="6"
-			height="5.5"
-			x="11"
-			y="14.5"
-			stroke="#000"
-			strokeWidth="3"
-			mask="url(#b)"
+			x="12.5"
```
Comment (danieldudzic) [inline assets/js/blocks/product-gallery/icon.tsx]:
```suggestion
			x="12.25"
```

**Review by danieldudzic — CHANGES_REQUESTED** [review]
There seems to be an overcorrection now.

Single_Product_‹_Template_‹_Editor_‹_productgallery_—_WordPress

**Comment by thealexandrelara** [comment]
@danieldudzic Thank you for the suggested values, I just committed the suggested change 🙌

**Review by gigitux — APPROVED** [review]
image

The image is still not full aligned, but we can iterate it in the next weeks. @thealexandrelara can you open a follow-up issue?

LGTM

**Comment by thealexandrelara** [comment]
> image
> The image is still not full aligned, but we can iterate it in the next weeks. @thealexandrelara can you open a follow-up issue?
> 
> LGTM

For me, it's correctly aligned. In your image, the stroke width is narrower than the current version (in which the stroke for the rectangle symbols matches the stroke for the "painting" symbol); maybe it is a cache issue?

image

**Comment by gigitux** [comment]
Mmm, I downloaded the SVG locally and opened with inkspace and it seems that the image isn't aligned 🤔 

How do you check if the image is aligned?

**Comment by thealexandrelara** [comment]
> Mmm, I downloaded the SVG locally and opened with inkspace and it seems that the image isn't aligned 🤔
> 
> How do you check if the image is aligned?

I just pasted the SVG code in https://www.svgviewer.dev/ and in there the icon is misaligned, like in your image:
image

However, when I tested it in the browsers, the icon is correctly aligned. 

Here is in Chrome:

image

Here is in Firefox:

image

Here is in Safari:

image

I don't know exactly why this is happening, but it seems browsers handle SVGs in a different way 🤔

**Pull request merged**